### PR TITLE
Fix 6483 do not download binary for conan info

### LINKED ISSUE #6483: [bug] Slow package resolution from local cache when remotes are defined
Context:
Recently we're having issues with our network infrastructure and our internal Conan remotes. In general it takes up to 5 seconds for a small package to get downloaded. We have a CI job which installs a Conan package and then does a number of  `conan info` and `conan inspect` on it.

Issue:
Because of this we found the issue that even though packages are available in the local Conan cache it takes a long time to access them when remotes are defined.

Expected:
* First `conan install A/1.0.0@user/channel` takes 5sec because of the slow remotes
* Further calls to `A/1.0.0@user/channel` are very fast because the package is already available in the local Conan Cache

Actual experience:
* First `conan install A/1.0.0@user/channel` takes 5sec because of the slow remotes
* Further calls take very long as well, even when the final package is taken from local Conan cache anyways
* We do not define -U parameters anywhere, so it should not query any data from the remotes

Some strange side effects:
* This happens when there are any remotes configured at all. Doesn't matter if they're all deactivated as well.
* This does not happen when there are no remotes configured. Then the package is taken from the local Conan cache in no time.

### Environment Details (include every applicable attribute)
  * Operating System+version: Dockerized build-environment based on Linux Suse 12.2 and 13.1
  * Compiler+version: -
  * Conan version: 1.21.0
  * Python version: Python 3.7

### Steps to reproduce (Include if Applicable)
Repeat
`time conan info <ref>`
a few times. Note the execution times.
Then do a 
`conan remote clean`
Repeat 
`time conan info <ref>`
a few times. Note the execution times, which are now faster than with remotes defined.
Redefine the remote(s)
and do the 
`time conan info <ref>`
again. Times are now slower again.
Times are still slow, when all remotes are disabled via
'conan remote disable <remote>'

### PR DESCRIPTION
**Pull request by monsdar**
Changelog: Fix: Do not download binary data during conan info when no remote is explicitly given by the user. Fixes #6483 
Docs: Not necessary??

- [x] Refer to the issue that supports this Pull Request.
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [ ] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. 

<sup>**Note:** By default this PR will skip the slower tests and will use a limited set of python versions. Check [here](https://github.com/conan-io/conan/blob/develop/.github/PR_INCREASE_TESTING.md) how to increase the testing level by writing some tags in the current PR body text.</sup>

### CODE REVIEW

**Comment by monsdar** [comment]
Seems the Git Log got messed up from within my fork... The older commits listed are all changes already merged into Conan or trivial (me updating my fork to the latest version of Conan). In one case I used a different Username (my real name), so it looks like two users have worked on this PR -.-

File: conans/client/graph/graph_binaries.py [hunk 1]
```diff
@@ -98,9 +98,9 @@ def _evaluate_cache_pkg(self, node, package_layout, pref, metadata, remote, remo
 
     def _evaluate_remote_pkg(self, node, pref, remote, remotes):
         remote_info = None
-        if remote:
+        if remotes.selected:
```
Comment (memsharded) [inline conans/client/graph/graph_binaries.py]: This is breaking most of the binary-remote logic in Conan.

If no remote is explicitly selected, the location of the binary package is:
- The one defined from the binary package, if it was previously downloaded
- The one defined for the recipe, if the recipe is associated with a remote.

As you can see in the CI, a ton of tests are broken by this.

We need to understand better:
- If this is a regression, it seems it is not, the behavior has always been there
- I'd say that the binary is not really downloaded, but the server is actually queried for the information of it. That is what the ``conan info`` command does. The title might be a bit misleading.
- Disabling all remotes indeed should completely avoid contacting them, if true, that could be a bug, but not related to the ``remotes.selected`` specification of remote.

**Review by memsharded — CHANGES_REQUESTED** [review]

**Comment by jgsogo** [comment]
@monsdar Thanks for trying to address the performance issue you detected in https://github.com/conan-io/conan/issues/6483 It looks like it requires a little bit more investigation, maybe, instead of using the `remotes.selected` it is about checking the `activated/deactivated` flag associated to them

**Comment by monsdar** [comment]
Yeah, saw that it was breaking a lot of things as well. Would've been too easy if it was nothing more than this easy fix.

The question is if `conan info` really needs to gather info about remote binaries or not.
* If it does, perhaps there could be a parameter to control whether to query remotes or not
* If it doesn't perhaps it's possible to internally avoid querying the remotes for binary data

**Comment by monsdar** [comment]
Let's rather continue the discussion within the Issue

**Comment by memsharded** [comment]
PR https://github.com/conan-io/conan/pull/12807 should close this PR, it has been a long time, and it didn't move forward.
https://github.com/conan-io/conan/pull/12807 contains a rationale of the current behavior, which seems correct and expected, and not hitting the servers if the packages are in cache, and only hitting if they are not in cache.

**Comment by memsharded** [comment]
Closed by https://github.com/conan-io/conan/pull/12807, please check also related https://github.com/conan-io/conan/pull/12808 (for Conan 2.0)

**Pull request closed without merging**